src/plugins/data/public/ui/dataset_selector/_dataset_explorer.scss

```diff
@@ -8,7 +8,7 @@
   grid-template-columns: repeat(auto-fit, minmax(200px, 240px)) minmax(300px, 1fr);
   height: 100%;
   max-height: calc(100vh - 200px);
-  overflow-x: auto;
+  overflow-x: hidden;
```

Comment:
I saw this issue several times but now not able to reproduce the issue but this setting may prevent it from scrolling. will `scrollbar-gutter: stable;` work?

Reply:
that works as well, but the scroll bar gutter will always present like this
![image](https://github.com/user-attachments/assets/cc182289-1916-44a0-9883-12dad330ca9c)

Reply:
To double check the selectable items, it has css for `text-overflow: eclipses`, so this probably will not have scrollbar for x-axis

![image](https://github.com/user-attachments/assets/58a91afd-212f-4433-b4ae-87c13295f18e)

Reply:
+1 to @SuZhou-Joe's suggestion. Disabling x scrollbars to address these reflow issues present an accessibility issue as the rightmost content can be no longer accessible for users with page zoom or smaller screen sizes.

Reply:
> text-overflow: eclipses
The 3rd column'd width is auto so text-overflow may not work for the 3rd column.

> the scroll bar gutter will always present like this
Yeah gutter will preserve some width for the scrollbar and I think we may need to make some trade-off to address the flashing issue.

Reply:
make sense, considering accessibility which is more important, let's me switch it to `scrollbar-gutter: stable;`

Reply:
when switch to `scrollbar-gutter: stable;` the issue still happened, let me convert it to draft first

Reply:
@SuZhou-Joe @d-buckner after more research, the issue happens when both x and y scrollbar present, to hide x scrollbar may have accessibility issue for small screen size. Instead, we can hide y scrollbar to avoid they both present causing flashing. 

Each column in the grid has its own y scrollbar, the remove outer one could remove double y scrollbar too.
![image](https://github.com/user-attachments/assets/3df423b0-0e74-4a11-9f87-eaa11d1225e5)

Reply:
Thanks @Hailong-am for trying on those options, looks good to me.

Reply:
Thanks for taking the time @Hailong-am! Changes look good to me as well.